### ISSUE TITLE
Module 'sys' has no attribute 'argv'

### ISSUE THREAD

**Issue by abalkin** [KxSystems/embedPy#48]
```
q)p)import sys;print(sys.argv)
'e: module 'sys' has no attribute 'argv'
  [0]  .../p.q:11: .p.e:{$["def"~3#x;$[x[3]in"<*>";ei 3;eo];"class"~5#x;$[x[5]in"*>";ei 5;eo];eo]x}
```

It would be nice to set sys.argv in p.q to a list derived from .z.X.

See https://docs.python.org/3.6/library/sys.html#sys.argv for details.

**Comment by jhanna-kx** [comment]
@abalkin would you suggest we just extend sys.argv with .z.X or include something to distinguish PyQ from embedPy?

**Comment by abalkin** [comment]
Good question.  In PyQ we populate sys.argv with enlist[string .z.f],.z.x with an additional trick to allow passing 1-letter options.  I don't think this logic is appropriate for embedPy which unlike pyq does not try to pretend being python.  In my own use cases, it did not really matter what sys.argv was set to as long as it was there.  IIRC, I had some module crash on import with an AttributeError.

This said, my suggestion is that sys.argv is set to [''] when embedPy is loaded interactively and to 1_.z.X when loaded from a script.

**Comment by jhanna-kx** [comment]
out of curiosity why shouldn't we just *always* set to `1_.z.X`

**Comment by abalkin** [comment]
According to the [documentation](https://docs.python.org/3/library/sys.html#sys.argv),

> `argv[0]` is the script name (it is operating system dependent whether this is a full pathname or not). If the command was executed using the -c command line option to the interpreter, `argv[0]` is set to the string '-c'. If no script name was passed to the Python interpreter, `argv[0]` is the empty string.

Of course, embedPy does not have to follow this convention exactly, but I think most python users expect `sys.argv[0]` to always be available.

A simple solution would be to ignore .z.x or .z.X and always set `sys.argv=['']`.  This is what interactive users expect and script writers will have to arrange to set  `sys.argv` before importing modules that use it.

**Comment by jhanna-kx** [comment]
Should be fixed in [1.2.0](../releases/1.2.0) release, we set sys.argv to [''] if it is not already set in p.q so if someone sets it themselves after loading `p.q` then subsequent loads of p.q wiil not overwrite it.
